PREFIXING: remove vendor prefixes (#19673)
mdn/content: files/en-us/glossary/vendor_prefix/index.md

  - CodingScripting
  - Glossary
---
Browser used to add prefixes to experimental or nonstandard CSS properties and JavaScript APIs, so developers could experiment with new ideas while—in theory—preventing their experiments from being relied upon and then breaking web developers' code during the standardization process. 

> Web developers included prefixed features on production Web sites, despite their experimental nature. This made it more difficult for browser vendors to ensure compatibility while working on new features. Including prefixed features also harmed smaller browsers who ended up having to add other browsers' prefixes in order to load popular web sites.

Reviewer: ```suggestion
Web developers included prefixed features on production websites, despite their experimental nature. This made it more difficult for browser vendors to ensure compatibility while working on new features. Including prefixed features also harmed smaller browsers who ended up having to add other browsers' prefixes to load popular websites.
```

> Web developers included prefixed features on production websites, despite their experimental nature. This made it more difficult for browser vendors to ensure compatibility while working on new features. Including prefixed features also harmed smaller browsers who ended up having to add other browsers' prefixes to load popular websites.

Author: ```suggestion
Web developers included prefixed features on production websites, despite their experimental nature. This made it more difficult for browser vendors to ensure compatibility while working on new features. Including prefixed features also harmed smaller browser vendors who ended up having to add other browsers' prefixes to load popular websites.
```

> Web developers included prefixed features on production websites, despite their experimental nature. This made it more difficult for browser vendors to ensure compatibility while working on new features. Including prefixed features also harmed smaller browser vendors who ended up having to add other browsers' prefixes to load popular websites.

Author: ```suggestion
Web developers included prefixed features on production websites, despite their experimental nature. This made it more difficult for browser vendors to ensure compatibility while working on new features. Including prefixed features also harmed smaller browser vendors who ended up having to add other browsers' prefixes in order to render popular websites.
```

transition: all 4s ease;
```

If you encounter the above code in a code base, you can safely remove all but the last line. All browsers support [transitions](/en-US/docs/Web/CSS/transition#browser_compatibility) without vendor prefixes:

```css
transition: all 4s ease;
```

## API prefixes

> Historically, vendors have also used prefixes for experimental APIs. If an entire interface was experimental, then the interface's name was prefixed (but not the properties or methods within). If an experimental property or method was added to a standardized interface, then the individual method or property was prefixed.

Reviewer 2: ```suggestion
Historically, browser vendors have also used prefixes for experimental APIs. If an entire interface was experimental, then the interface's name was prefixed (but not the properties or methods within). If an experimental property or method was added to a standardized interface, then the individual method or property was prefixed.
```

                               window.msRequestAnimationFrame;
```

> If you encounter the above code in a code base, you can safely remove all but the first line. All browsers support [`requestAnimationFrame`](/en-US/docs/Web/API/window/requestAnimationFrame#browser_compatibility) without vendor prefixes:

Author: ```suggestion
If you encounter the above code in a code base, you can safely remove all but the first line. All browsers support [`requestAnimationFrame`](/en-US/docs/Web/API/window/requestAnimationFrame#browser_compatibility) without vendor prefixes, and without `window`:
```

```js

> window.requestAnimationFrame = window.requestAnimationFrame;

Reviewer: No need to declare it, not even a need to have `window.` in front of it.
```suggestion
requestAnimationFrame(callback);
```

```

## See also

> - [Vendor prefix](https://en.wikipedia.org/wiki/CSS_hack#Browser_prefixes) on Wikipedia

Reviewer: Should we rename this page?
